src/gui/widgets/multiline_text.cpp

```diff
@@ -405,7 +435,27 @@ void multiline_text::signal_handler_left_button_down(const event::ui_event event
 	get_window()->keyboard_capture(this);
 	get_window()->mouse_capture();
 
-	handle_mouse_selection(get_mouse_position(), true);
+	point mouse_pos = get_mouse_position();
+
+	if (get_link_aware()) {
+		if (desktop::open_object_is_supported()) {
+			std::string link = get_label_link(mouse_pos - get_origin());
```

Comment:
Out of curiosity, does `get_mouse_position` not respect the current viewport?

Reply:
It doesn't. Following the call hierarchy ultimately leads me to https://github.com/wesnoth/wesnoth/blob/ea8cdf92823c15e84aa05827a2e5b5ba9f291842/src/sdl/input.cpp#L27-L47.